pontoon/static/js/widgets/table.js

```diff
@@ -0,0 +1,153 @@
+            this.visible.length = 0;
+            this.setState({checked: new Set()});
+        }
+    }
+
+    handleCheckboxClick (evt) {
+        // adds/removes paths for submission
+        let {checked} = this.state;
+        checked = new Set(checked);
+        evt.target.checked ? checked.add(evt.target.name) : checked.delete(evt.target.name);
+        this.setState({checked});
+    }
+
+    handleSelectAll () {
+        // user clicked the select all checkbox...
+        // if there are some resources checked already, all are removed
+        // otherwise all visible are checked.
+        let {checked} = this.state;
+        checked = checked.size > 0 ? new Set() : new Set([...this.visible.filter(x => x)]);
+        return this.setState({checked});
+    }
+
+    async handleSubmit (evt) {
+        // after emitting handleSubmit to parent with list of currently
+        // checked, clears the checkboxes
+        evt.preventDefault();
+        const {checked} = this.state;
+        await this.props.handleSubmit({data: [...checked]});
+        await this.setState({checked: new Set()});
+    }
+
+    handleTableChange () {
+        this.visible.length = 0;
+        return this.setState({checked: new Set()});
+    }
+
+    handleTableResize (pageSize) {
+        this.visible.length = pageSize;
+        return this.setState({checked: this.pruned});
+    }
+
+    handleTableSortChange () {
+        return this.setState({checked: this.pruned});
+    }
+
+    renderCheckbox (data) {
+        const {checked} = this.state;
+        return (
+            <Checkbox
+               checked={checked.has(data.original[0])}
+               name={data.original[0]}
+               onChange={this.handleCheckboxClick} />);
+    }
+
+    render () {
+        return (
+            <div>
+              {this.renderTable()}
+              {this.renderSubmit()}
+            </div>);
+    }
+
+    renderSelectAllCheckbox () {
+        // renders a select all checkbox, sets the check to
+        // indeterminate if only some of the visible resources
+        // are checked
+        let {all, checked} = this.selected;
+        return (
+            <Checkbox
+               checked={all}
+               indeterminate={!all && checked.size > 0}
+               onClick={this.handleSelectAll} />);
+    }
+
+    renderSubmit () {
+        return (
+            <button
+               className={this.submitClass}
+               onClick={this.handleSubmit}>
+              {this.submitMessage}
+            </button>);
+    }
+
+    renderTable () {
+        return (
+            <ReactTable
+               defaultPageSize={this.defaultPageSize}
+               className="-striped -highlight"
+               data={this.props.data}
+               onPageChange={this.handleTableChange}
+               onPageSizeChange={this.handleTableResize}
+               onSortedChange={this.handleTableSortChange}
+               columns={this.columns} />);
```

Comment:
If I'm not mistaken, `this.columns` is undefined.

Reply:
hmm, yep, this was due to this component still being inherited from - i was on the way to changing that when splitting out this PR. @julen thanks for this review, its been really helpful